eng/test-mounts.sh

```diff
@@ -2,36 +2,91 @@
 set -xeuo pipefail
 IFS=$'\n\t'
 
+print_to_workflow() {
+    if [[ -n ${GITHUB_STEP_SUMMARY+x} ]]; then
```

Comment:
suggestion: Maybe it is just me, but I always have to look up what the [Bash Parameter Expansion](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html) syntax does. This should be equivalent without relying on bash parameter expansion.

```suggestion
    if [[ ! -z "${GITHUB_STEP_SUMMARY}" ]]; then
```

Reply:
The `+x` is a fail-safe to differentiate between the variable being unset and undefined. In this case, the condition will only evaluate to false if `GITHUB_STEP_SUMMARY` is unset. If it is set to any value, even an empty string, it will evaluate to true.

In retrospect, I don't think that would be necessary in this context, considering that an empty `GITHUB_STEP_SUMMARY` would be invalid anyway, so just removing `+x` should cut it.

Also, `[[ ! -z $variable ]]` and `[[ -n $variable ]]` are equivalent, although I usually avoid the double-negative [for stylistic reasons](https://www.shellcheck.net/wiki/SC2236).